### PR TITLE
Bug 2878 bitfinex partially filled problem

### PR DESCRIPTION
**Pull request by Adalyat**
#### Description

In order to Lean could handle order filling properly it's necessary to pass Filled status for the last trade update. As Bitfinex doesn't send us Filled Size within each trade update we have to track it manually

#### Related Issue

#2878 

#### Motivation and Context

LEAN can't handle order filled by multiple trades if no Filled status was passed in event

#### How Has This Been Tested?

Run brokerage based algorithm and check all executed orders including fully or in multiple pieces.

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs [hunk 1]
```diff
@@ -577,7 +570,21 @@ private void EmitFillOrder(string[] entries)
...
+                        filling[order.Id] += fillQuantity;
+                    }
+                    else
+                    {
+                        filling.AddOrUpdate(order.Id, fillQuantity);
+                    }
+                    status = filling[order.Id] == order.Quantity
+                        ? OrderStatus.Filled
+                        : OrderStatus.PartiallyFilled;
+                }
```
Comment (StefanoRaggi) [inline Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs]: We can reduce the number of dictionary accesses here, something like this:
``` C#
                if (fillQuantity != order.Quantity)
                {
                    decimal totalFillQuantity;
                    filling.TryGetValue(order.Id, out totalFillQuantity);
                    totalFillQuantity += fillQuantity;
                    filling[order.Id] = totalFillQuantity;

                    status = totalFillQuantity == order.Quantity
                        ? OrderStatus.Filled
                        : OrderStatus.PartiallyFilled;
                }
```

File: Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs [hunk 2]
```diff
@@ -589,11 +596,12 @@ private void EmitFillOrder(string[] entries)
                 // if the order is closed, we no longer need it in the active order list
                 lock (closeLocker)
```
Comment (StefanoRaggi) [inline Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs]: I think we can safely remove the `lock` statement and the `closeLocker` variable

**Comment by voidale** [comment]
Thanks for this! Will make live trading possible :)

File: Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs (hunk 1 above)
Comment (voidale) [reply Brokerages/Bitfinex/BitfinexBrokerage.Messaging.cs]: @StefanoRaggi It looks like it's now updated can we get it merged?

**Pull request merged**